llama-index-core/llama_index/core/workflow/context.py

```diff
@@ -40,6 +83,83 @@ def __init__(self, workflow: "Workflow", stepwise: bool = False) -> None:
         # Step-specific instance
         self._events_buffer: Dict[Type[Event], List[Event]] = defaultdict(list)
 
+        # keep track of all the event classes that are accepted by the workflow
+        self._event_classes: Dict[str, Type[Event]] = {}
+        for step_func in self._workflow._get_steps().values():
```

Comment:
Instead of keeping a list of class objects, we could serialize a class with its qualified name (see https://github.com/deepset-ai/haystack/blob/main/haystack/core/serialization.py#L74)  that later we can pass to `importlib` with something like https://github.com/deepset-ai/haystack/blob/main/haystack/core/serialization.py#L195

Reply:
interesting, will take a look!

Reply:
I have an approach working with this... slightly concerned about security issues, but I suppose if you are importing something already on your machine, and thats an issue for you, you have other problems 😅